gitrevise/tui.py

```diff
@@ -172,6 +185,9 @@ def noninteractive(
     if args.cut:
         current = cut_commit(current)
 
+    if repo.sign_commits and not hasattr(current, "gpgsig"):
+        current = sign_commit(current)
+
```

Comment:
I think the best way to do this would probably be to add a new `recommit` option to `commit.update(...)` which skips the `unchanged` check. Then this could be written as:

```python
if repo.sign_commits and not current.gpgsig:
    current = current.update(recommit=True)
```